ports/samd/mpconfigport.h

```diff
@@ -35,7 +35,6 @@
 #define MICROPY_GC_STACK_ENTRY_TYPE         uint16_t
 #define MICROPY_GC_ALLOC_THRESHOLD          (0)
 #define MICROPY_ALLOC_PATH_MAX              (256)
-#define MICROPY_QSTR_BYTES_IN_HASH          (1)
```

Comment:
Maybe this should stay, to keep SAMD size down?  Or maybe move it to `samd/mcu/samd21/mpconfigmcu.h`?

Reply:
SAMD doesn't set the feature level, so this is the default anyway.

Reply:
But the samd port has increased in size, so this change is not a no-op.  We need to make a conscious decision about whether the current config for samd should be changed, or not.

Reply:
Right. I missed that it's set in the `samd/mcu/*/mpconfigmcu.h`...

I think it makes sense then for samd21 to be 1 and samd51 to be 2. i.e. d21 is unchanged, but d51 is now moving to 2.

Reply:
Updated the commit message to match.

Reply:
@jimmo I made the cross-check with the SAMD ItsyBitsy M4 build. Setting `#define MICROPY_QSTR_BYTES_IN_HASH          (0)` in mpconfigport,h **reduced** the flash size by 2064 bytes. For the ItsyBitsy M0, the size **decrease** is 684 bytes.

Reply:
@robert-hh I think all those results sound expected. Thanks for checking. Just to confirm, I expect from this PR:
- D21 boards are unchanged and should have no firmware size change.
- D51 boards should grow by roughly 1kiB (and get a small performance boost).

> For these, setting the QSTR hash to 0 is an advantage.

I don't think we should set bytes-in-hash to zero by default for anything except the "true" minimal ports/variants (where we're really trying to highlight the minimal possible configuration), or unless we're at the size limit (for example some of those stm32x0 boards). As you've noted, the performance impact is non-trivial.